### PR TITLE
remove `driver_info` class from `dbGetInfo()` output

### LINKED ISSUE #667: Eliminate `driver_info` class
It looks vestigial and I don't think it does anything any more.

### PR DESCRIPTION
**Pull request by simonpcouch**
Closes #667.

Introduced in https://github.com/r-dbi/odbc/commit/9585993b6eeb42621eaebd6029dd9cfe63535400. It didn't seem to have had any methods attached to it then, either.

### CODE REVIEW

File: R/dbi-connection.R [hunk 1]
```diff
@@ -220,7 +220,7 @@ setMethod("dbListFields", c("OdbcConnection", "character"),
 setMethod("dbGetInfo", "OdbcConnection",
   function(dbObj, ...) {
     info <- connection_info(dbObj@ptr)
-    structure(info, class = c(info$dbms.name, "driver_info", "list"))
```
Comment (hadley) [inline R/dbi-connection.R:223]: Since it's now just a "list", you can yeet the class attribute 😄 (and same below)

Comment (simonpcouch) [reply R/dbi-connection.R:223]: So the `info$dbms.name` class is unneeded as well?

Comment (hadley) [reply R/dbi-connection.R:223]: Oh hmmm, that's a good question. But I don't think so since that's usually an S4 class and here we're making an S3 class.

Comment (hadley) [reply R/dbi-connection.R:223]: I double checked this by looking at the `NAMESPACE` to see what S3 generics we have methods for. As far as I can tell, the only S3 generic is `odbcDataType()`, and that's always called on the connection object.

**Review by hadley — APPROVED** [review]
This PR slaps.

**Comment by simonpcouch** [comment]
😂😂😂

**Pull request merged**